pytorch_lightning/metrics/functional/classification.py

```diff
@@ -900,6 +901,62 @@ def _auroc(pred, target, sample_weight, pos_label):
     return _auroc(pred=pred, target=target, sample_weight=sample_weight, pos_label=pos_label)
 
 
+def multiclass_auroc(
+        pred: torch.Tensor,
+        target: torch.Tensor,
+        sample_weight: Optional[Sequence] = None,
+        num_classes: Optional[int] = None,
+) -> torch.Tensor:
+    """
+    Compute Area Under the Receiver Operating Characteristic Curve (ROC AUC) from multiclass
+    prediction scores
+
+    Args:
+        pred: estimated probabilities, with shape [N, C]
+        target: ground-truth labels, with shape [N,]
+        sample_weight: sample weights
+        num_classes: number of classes (default: None, computes automatically from data)
+
+    Return:
+        Tensor containing ROCAUC score
+
+    Example:
+
+        >>> pred = torch.tensor([[0.85, 0.05, 0.05, 0.05],
+        ...                      [0.05, 0.85, 0.05, 0.05],
+        ...                      [0.05, 0.05, 0.85, 0.05],
+        ...                      [0.05, 0.05, 0.05, 0.85]])
+        >>> target = torch.tensor([0, 1, 3, 2])
+        >>> multiclass_auroc(pred, target)   # doctest: +NORMALIZE_WHITESPACE
+        tensor(0.6667)
+    """
+    if not torch.allclose(pred.sum(dim=1), torch.tensor(1.0)):
+        raise ValueError(
+            "Multiclass AUROC metric expects the target scores to be"
+            " probabilities, i.e. they should sum up to 1.0 over classes")
+
+    if torch.unique(target).size(0) != pred.size(1):
```

Comment:
Shouldn't it be `torch.unique(target).size(0) <= pred.size(1)`

```
target = torch.tensor([0, 0, 0, 0])
>>> torch.unique(target).size(0)
1
```

Reply:
Or we could have a get_num_classes utils too.

Reply:
Well the metric is undefined when `torch.unique(target).size(0) < pred.size(1)`, that's why there is a strict equals. The way this implementation works (based on sklearn's version) is, that it uses a one-vs-rest strategy of computing the AUROC. For `n` classes it computes `n` binary AUROCs, for each class in turn, it is considered a positive class and all the other classes negative. Then it averages those.

E.g., for `n=3` and `target=[0, 1, 1, 2]`, for class `0` we binarize the target to make `0` the positive class: `[1, 0, 0, 0]` and compute the `AUC` of `ROC` of that.

If a target label is not present in the `target`, e.g. `n=3` and `target=[0, 1, 1, 1]` then for the absent class `2` the binarized target would look like `[0, 0, 0, 0]` (all negative) and `ROC` cannot be computed (would raise an error). Consequently, the whole multiclass AUROC is undefined in that case.

Reply:
Make sense !